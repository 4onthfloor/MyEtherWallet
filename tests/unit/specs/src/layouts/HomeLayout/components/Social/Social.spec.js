import { shallowMount } from '@vue/test-utils';
import Social from '@/layouts/HomeLayout/components/Social/Social.vue';
import { Tooling } from '@@/helpers';

describe('Social.vue', () => {
  let localVue, i18n, wrapper, store;

  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;
  });

  beforeEach(() => {
    wrapper = shallowMount(Social, {
      localVue,
      i18n,
      store,
      attachToDocument: true
    });
  });

  it('should have correct data', () => {
    expect(wrapper.vm.$data.links.length).toBe(5);
  });
  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('should render correct contents', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should dismount component', () => {
    wrapper.destroy();
    expect(wrapper.exists()).toBe(false);
  });
});
