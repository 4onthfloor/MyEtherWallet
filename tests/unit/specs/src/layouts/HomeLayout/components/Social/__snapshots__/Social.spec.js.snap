// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social.vue should render correct contents 1`] = `
<div class="social">
  <div class="wrap">
    <div class="page-container">
      <div class="flex-col-vertical-center content">
        <div class="title">
          <h2>Join MEW Community</h2>
        </div>
        <div class="icons"><a href="https://www.facebook.com/MyEtherWallet" aria-label="Facebook" target="_blank" rel="noopener noreferrer">
            <div class="Facebook"><img src="" alt="">
              <p>Facebook</p>
            </div>
          </a><a href="https://twitter.com/myetherwallet" aria-label="Twitter" target="_blank" rel="noopener noreferrer">
            <div class="Twitter"><img src="" alt="">
              <p>Twitter</p>
            </div>
          </a><a href="https://medium.com/@myetherwallet" aria-label="Medium" target="_blank" rel="noopener noreferrer">
            <div class="Medium"><img src="" alt="">
              <p>Medium</p>
            </div>
          </a><a href="https://www.reddit.com/r/myetherwallet" aria-label="reddit" target="_blank" rel="noopener noreferrer">
            <div class="reddit"><img src="" alt="">
              <p>Reddit</p>
            </div>
          </a><a href="https://www.github.com/myetherwallet" aria-label="github" target="_blank" rel="noopener noreferrer">
            <div class="github"><img src="" alt="">
              <p>Github</p>
            </div>
          </a></div>
      </div>
    </div>
  </div>
</div>
`;
